File: lab1-3-csharp/Book.cs
namespace BookLibrary
{
    /// <summary>
    /// Represents a book in the library system
    /// </summary>
    public class Book
    {
        public string Title { get; set; }
        public string Author { get; set; }
        public string ISBN { get; set; }
        public int PublicationYear { get; set; }
        public bool IsAvailable { get; set; }

        public Book(string title, string author, string isbn, int publicationYear)
        {
            Title = title;
            Author = author;
            ISBN = isbn;
            PublicationYear = publicationYear;
            IsAvailable = true;
        }

        public override string ToString()
        {
            string availability = IsAvailable ? "Available" : "Checked Out";
            return $"\"{Title}\" by {Author} ({PublicationYear}) - ISBN: {ISBN} - {availability}";
        }
    }
}

File: lab-typescript/src/book.ts
/**
 * Represents a book in the library system
 */
export class Book {
  public title: string;
  public author: string;
  public isbn: string;
  public publicationYear: number;
  public isAvailable: boolean;

  constructor(
    title: string,
    author: string,
    isbn: string,
    publicationYear: number
  ) {
    this.title = title;
    this.author = author;
    this.isbn = isbn;
    this.publicationYear = publicationYear;
    this.isAvailable = true;
  }

  toString(): string {
    const availability = this.isAvailable ? "Available" : "Checked Out";
    return `"${this.title}" by ${this.author} (${this.publicationYear}) - ISBN: ${this.isbn} - ${availability}`;
  }
}
